skbuild/resources/cmake/FindPythonExtensions.cmake

```diff
@@ -317,6 +317,37 @@ if(NOT DEFINED PYTHON_EXTENSION_MODULE_SUFFIX)
   set(PYTHON_EXTENSION_MODULE_SUFFIX "${_item}")
 endif()
 
+function(_set_python_extension_symbol_visibility _target)
+  if(PYTHON_VERSION_MAJOR VERSION_GREATER 2)
+    set(_modinit_prefix "PyInit_")
+  else()
+    set(_modinit_prefix "init")
+  endif()
+  message("_modinit_prefix:${_modinit_prefix}")
+  if("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
+    set_target_properties(${_target} PROPERTIES LINK_FLAGS 
+        "/EXPORT:${_modinit_prefix}${_target}"
+    )
+  elseif("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU")
+    set(_script_path
+      ${CMAKE_CURRENT_BINARY_DIR}/_skbuild/${_target}-version-script.map
+    )
+    file(WRITE
+      ${_script_path}
+      "
+      {
+        global:
+          ${_modinit_prefix}${_target};
+        local: *;
+      }
+      "      
+    )
+    set_target_properties(${_target} PROPERTIES LINK_FLAGS 
```

Comment:
It turns out that this a compile flags and not a link flags. `COMPILE_FLAGS` should be used

Reply:
On windows, link flags are used (I think).

Reply:
I talked to fast ... in fact the following is happening:
* `_set_python_extension_symbol_visibility` set the linker flags as it should 
* but these are reset to `-Wl,--unresolved-symbols=ignore-all` when gnu is used. See function `target_link_libraries_with_dynamic_lookup` called lower in this file. 

I will try to call the function `_set_python_extension_symbol_visibility` afterward

Reply:
Now that the linker is flag is passed, here is the error:

```
[100%] Linking CXX shared module _hello.cpython-35m-x86_64-linux-gnu.so
cd /tmp/pytest-of-jcfr/pytest-jcfr/_skbuild/linux-x86_64-3.5/cmake-build/hello && /home/jcfr/Software/cmake-3.10.0-Linux-x86_64/bin/cmake -E cmake_link_script CMakeFiles/_hello.dir/link.txt --verbose=1
/usr/bin/c++ -fPIC -O3 -DNDEBUG -Wl,--version-script=/tmp/pytest-of-jcfr/pytest-jcfr/_skbuild/linux-x86_64-3.5/cmake-build/hello/_skbuild/_hello-version-script.map -shared  -o _hello.cpython-35m-x86_64-linux-gnu.so CMakeFiles/_hello.dir/_hello.cxx.o 
/usr/bin/ld:/tmp/pytest-of-jcfr/pytest-jcfr/_skbuild/linux-x86_64-3.5/cmake-build/hello/_skbuild/_hello-version-script.map:7: syntax error in VERSION script
collect2: error: ld returned 1 exit status
hello/CMakeFiles/_hello.dir/build.make:94: recipe for target 'hello/_hello.cpython-35m-x86_64-linux-gnu.so' failed
make[2]: *** [hello/_hello.cpython-35m-x86_64-linux-gnu.so] Error 1

```

Reply:
If that is the case, then it's probably best to append the version script to the `-Wl,--unresolved-symbols=ignore-all` argument.